core/src/main/java/org/apache/james/core/healthcheck/Result.java

```diff
@@ -26,25 +26,66 @@
 public class Result {
 
     public static Result healthy(ComponentName componentName) {
-        return new Result(componentName, ResultStatus.HEALTHY, Optional.empty());
+        return Builder.builder().componentName(componentName).status(ResultStatus.HEALTHY).build();
     }
 
     public static Result unhealthy(ComponentName componentName, String cause) {
-        return new Result(componentName, ResultStatus.UNHEALTHY, Optional.of(cause));
+        return Builder.builder().componentName(componentName).status(ResultStatus.UNHEALTHY).cause(cause).build();
+    }
+
+    public static Result unhealthy(ComponentName componentName, String cause, Throwable error) {
+        return Builder.builder().componentName(componentName).status(ResultStatus.UNHEALTHY).cause(cause).error(error).build();
     }
 
     public static Result degraded(ComponentName componentName, String cause) {
-        return new Result(componentName, ResultStatus.DEGRADED, Optional.of(cause));
+        return Builder.builder().componentName(componentName).status(ResultStatus.DEGRADED).cause(cause).build();
+    }
+
+    public static class Builder {
+        private ComponentName componentName;
+        private ResultStatus status;
+        private String cause;
+        private Optional<Throwable> error = Optional.empty();
+
+        public Builder componentName(ComponentName componentName) {
+            this.componentName = componentName;
+            return this;
+        }
+
+        public Builder status(ResultStatus status) {
+            this.status = status;
+            return this;
+        }
+
+        public Builder cause(String cause) {
+            this.cause = cause;
+            return this;
+        }
+
+        public Builder error(Throwable error) {
+            this.error = Optional.of(error);
+            return this;
+        }
+
+        public Result build() {
+            return new Result(componentName, status, cause, error);
+        }
+
+        public static Builder builder() {
+            return new Builder();
+        }
     }
 
     private final ComponentName componentName;
     private final ResultStatus status;
-    private final Optional<String> cause;
+    private final String cause;
+    private final Optional<Throwable> error;
 
-    private Result(ComponentName componentName, ResultStatus status, Optional<String> cause) {
+    public Result(ComponentName componentName, ResultStatus status, String cause, Optional<Throwable> error) {
```

Comment:
why do you want to constructor to be public ?

Reply:
idem